prime-angular-frontend/src/app/modules/site-registration/pages/site-management-page/site-management-page.component.ts

```diff
@@ -193,23 +207,32 @@ export class SiteManagementPageComponent implements OnInit {
   }
 
   public ngOnInit(): void {
-    this.getOrganizations();
+    this.getOrganizations(0);
   }
 
-  private getOrganizations(): void {
+  /**
+   *  Get organization data and bind them to this component
+   * @param orgId organization Id, put zero when no specific organization id
+   */
+  private getOrganizations(orgId: number): void {
     this.busy = this.authService.getUser$()
       .pipe(
         exhaustMap((user: BcscUser) =>
           this.organizationResource.getSigningAuthorityOrganizationByUsername(user.username)
         ),
-        map((organization: Organization) => {
+        map((organizations: Organization[]) => {
+          const organization = orgId > 0 ?
+            organizations.find((org) => org.id === orgId)
+            : organizations[0];
+
           this.organizationSitesExpiryDates = organization.sites
             .reduce((expiryDates: string[], site: Site) => {
               return (site.status === SiteStatusType.EDITABLE && !!site.approvedDate)
                 ? [...expiryDates, Site.getExpiryDate(site)]
                 : expiryDates;
             }, []);
-          return this.organization = organization;
+          this.organizations = organizations;
+          return this.organization = organizations.find((org) => org.id === this.organizationService.organization.id);
```

Comment:
This seems to work but shouldn't this be `organizations.find((org) => org.id === organization.id);` instead?